hadoop-yarn-project/hadoop-yarn/hadoop-yarn-server/hadoop-yarn-server-resourcemanager/src/main/java/org/apache/hadoop/yarn/server/resourcemanager/rmapp/attempt/RMAppAttemptMetrics.java

```diff
@@ -242,4 +253,149 @@ public Resource getApplicationAttemptHeadroom() {
   public void setApplicationAttemptHeadRoom(Resource headRoom) {
     this.applicationHeadroom = headRoom;
   }
+
+  /**
+   * Add allocationID latency to the application ID with a timestap =
+   * CurrentTime (guaranteed)
+   *
+   * @param allocId the allocation Id to add If the allocation ID is already
+   *                present (which shouldn't happen) it ignores the entry
+   */
+  public void addAllocationGuarLatencyIfNotExists(long allocId) {
+    addAllocationGuarLatencyIfNotExists(allocId, System.currentTimeMillis());
+  }
+
+  /**
+   * Add allocationID latency to the application ID with a specific timestamp
+   * (guaranteed)
+   *
+   * @param allocId   allocationId
+   * @param timestamp the timestamp to associate
+   */
+  public void addAllocationGuarLatencyIfNotExists(long allocId,
+      long timestamp) {
+    allocationGuaranteedLatencies.putIfAbsent(allocId, timestamp);
+  }
+
+  /**
+   * Add allocationID latency to the application ID with a timestap =
+   * CurrentTime (opportunistic)
+   *
+   * @param allocId the allocation Id to add If the allocation ID is already
+   *                present (which shouldn't happen) it ignores the entry
+   */
+  public void addAllocationOppLatencyIfNotExists(long allocId) {
+    this.addAllocationOppLatencyIfNotExists(allocId,
+        System.currentTimeMillis());
+  }
+
+  /**
+   * Add allocationID latency to the application ID with a specific timestamp
+   * (opportunistic)
+   *
+   * @param allocId   allocationId
+   * @param timestamp the timestamp to associate
+   */
+  public void addAllocationOppLatencyIfNotExists(long allocId, long timestamp) {
+    allocationOpportunisticLatencies.putIfAbsent(allocId, timestamp);
+  }
+
+  /**
+   * Returns the time associated when the allocation Id was added This method
+   * removes the allocation Id from the class (guaranteed)
+   *
+   * @param allocId the allocation ID to get the associated time
+   * @return the timestamp associated with that allocation id as well as stop
+   * tracking it
+   */
+  public long getAndRemoveGuaAllocationLatencies(long allocId) {
+    Long ret = allocationGuaranteedLatencies.remove(new Long(allocId));
+    return ret != null ? ret : 0l;
+  }
+
+  /**
+   * Returns the time associated when the allocation Id was added This method
+   * removes the allocation Id from the class (opportunistic)
+   *
+   * @param allocId the allocation ID to get the associated time
+   * @return the timestamp associated with that allocation id as well as stop
+   * tracking it
+   */
+  public long getAndRemoveOppAllocationLatencies(long allocId) {
+    Long ret = allocationOpportunisticLatencies.remove(new Long(allocId));
```

Comment:
There's a deprecation and I think it might be enough to pass allocId without creating it.